### PR TITLE
Test for patching cluster resource replica count and domain resource replica count

### PR DESCRIPTION
**Pull request by maggiehe00**
Kind new
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/13323/

### CODE REVIEW

**Comment by maggiehe00** [comment]
@xiancao I'm addressing some review comments from Vanaja. Will ping you once it is done. Thanks.

**Comment by maggiehe00** [comment]
This PR is back to review.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java [hunk 1]
```diff
@@ -578,6 +582,58 @@ void testAdminServiceAnnotationsLabels() {
         "Missing expected label on admin service");
   }
 
+  /** The default replicas at domain resource level is 1. Patching domain1 with new replicas 3.
+   * Verify managed server pods count at cluster resource level are not affected by domain resource
+   * level patching.
+   * Replicas count at cluster resource level is defined as 2.
+   * Scale up the cluster by patching replicas count to 5 at cluster resource level.
+   * Scale down the cluster by patching replicas count to 1 at cluster resource level.
+   * Verify cluster resource level scaling is successful even though replicas count at domain resource
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java:591]: The first line of description says "replicas at domain resource level is 1"

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java [hunk 2]
```diff
@@ -578,6 +582,58 @@ void testAdminServiceAnnotationsLabels() {
...
+    checkPodDoesNotExist(managedServerPrefix + 3, domainUid, domainNamespace);
+
+
+    //scaling by patching replicas at cluster resource level should scale up/down
+    //the cluster even the domain resource has replicas defined
+    logger.info("Updating the cluster {0} replica count to 5", clusterName);
+    boolean p1Success = scaleCluster(clusterName, domainNamespace,5);
+    assertTrue(p1Success,
+        String.format("Patching replica to 5 failed for cluster %s in namespace %s",
+            clusterName, domainNamespace));
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java:628]: Need to check the existence of 5 manged servers to ensure the scale up is successful.

Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java:628]: add a check for 5 managed servers are up

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java [hunk 3]
```diff
@@ -578,6 +582,58 @@ void testAdminServiceAnnotationsLabels() {
...
+    logger.info("Updating the cluster {0} replica count to 1", clusterName);
+    p1Success = scaleCluster(clusterName, domainNamespace,1);
+    assertTrue(p1Success,
+        String.format("Patching replica to 1 failed for cluster %s in namespace %s",
+            clusterName, domainNamespace));
+
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java:634]: add check for managed servers, only 1 server should be running

**Review by vanajamukkara — CHANGES_REQUESTED** [review]

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java [hunk 4]
```diff
@@ -578,6 +582,58 @@ void testAdminServiceAnnotationsLabels() {
...
+    logger.info("Updating the cluster {0} replica count to 5", clusterName);
+    boolean p1Success = scaleCluster(clusterName, domainNamespace,5);
+    assertTrue(p1Success,
+        String.format("Patching replica to 5 failed for cluster %s in namespace %s",
+            clusterName, domainNamespace));
+    logger.info("Updating the cluster {0} replica count to 1", clusterName);
+    p1Success = scaleCluster(clusterName, domainNamespace,1);
+    assertTrue(p1Success,
+        String.format("Patching replica to 1 failed for cluster %s in namespace %s",
+            clusterName, domainNamespace));
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java:633]: Need to check all the managed servers are shutdown except one.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java (hunk 1 above)
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java:591]: Yes, that is default seting

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java (hunk 2 above)
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java:628]: done

Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java:628]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java (hunk 1 above)
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java:591]: Yes, that is the default setting.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java (hunk 4 above)
Comment (maggiehe00) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java:639]: done

**Review by anpanigr — APPROVED** [review]

**Pull request merged**